plugins/module_utils/network/sonic/config/ntp/ntp.py

```diff
@@ -0,0 +1,550 @@
+                        self._module.fail_json(msg=str(exc), code=exc.code)
+            result['changed'] = True
+        result['commands'] = commands
+
+        changed_ntp_facts = self.get_ntp_facts()
+
+        result['before'] = existing_ntp_facts
+        if result['changed']:
+            result['after'] = changed_ntp_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_ntp_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        if want is None:
+            want = []
+
+        have = existing_ntp_facts
+
+        resp = self.set_state(want, have)
+
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        state = self._module.params['state']
+
+        self.preprocess_want(want, state)
+
+        if state == 'deleted':
+            commands, requests = self._state_deleted(want, have)
+        elif state == 'merged':
+            commands, requests = self._state_merged(want, have)
+
+        return commands, requests
+
+    def _state_merged(self, want, have):
+        """ The command generator when state is merged
+
+        :param want: the additive configuration as a dictionary
+        :param obj_in_have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        diff = get_diff(want, have, TEST_KEYS_ADD)
+
+        commands = diff
+        requests = []
+        if commands:
+            requests = self.get_create_requests(commands, have)
+
+        if len(requests) > 0:
+            commands = update_states(commands, "merged")
+        else:
+            commands = []
+
+        return commands, requests
+
+    def _state_deleted(self, want, have):
+        """ The command generator when state is deleted
+
+        :param want: the objects from which the configuration should be removed
+        :param obj_in_have: the current configuration as a dictionary
+        :param interface_type: interface type
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        diff = get_diff(want, have, TEST_KEYS_DEL)
+
```

Comment:
Discussed with Kerry: update later.

Reply:
It looks like that a bug for which a JIRA should be filed. If you agree, please submit a bug report for it.

In the meantime, it would still be more efficient to handle a "delete_all" by deleting the keys, then doing a top level "delete_all".

Reply:
Added work-around.

Reply:
Verified.